### PR TITLE
Clean page_builder and added component spacing

### PR DESCRIPTION
**Pull request by JimmyHoenderdaal**

### CODE REVIEW

**Comment by JimmyHoenderdaal** [comment]
I separated the pagebuilder into fieldsets, update some frontend packages and added component spacing

File: resources/fieldsets/call_to_action_alt.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    handle: title
+    field:
+      buttons:
+        - h1
+        - h2
+        - h3
+        - h4
+        - h5
+        - h6
+        - bold
```
Comment (Roene-JustBetter) [inline resources/fieldsets/call_to_action_alt.yaml:13]: Do we need all of this for the title? I think h1/h6 will be enough?

File: resources/fieldsets/image_text.yaml [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+                  handle: title
+                  field:
+                    buttons:
+                      - h1
+                      - h2
+                      - h3
+                      - h4
+                      - h5
+                      - h6
+                      - bold
```
Comment (Roene-JustBetter) [inline resources/fieldsets/image_text.yaml:96]: Same as above. Do we all need this for the title?

File: resources/fieldsets/image_text.yaml [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+                  handle: subtitle
+                  field:
+                    buttons:
+                      - h1
+                      - h2
+                      - h3
+                      - h4
+                      - h5
+                      - h6
+                      - bold
```
Comment (Roene-JustBetter) [inline resources/fieldsets/image_text.yaml:145]: Same here.

File: resources/views/page-builder/featured_cards.blade.php [hunk 4]
```diff
@@ -12,8 +12,8 @@
     <div class="relative mt-7 w-full">
         <div class="absolute left-0 aspect-square w-[500px] -translate-x-1/2 -translate-y-1/2 rounded-full bg-cyan-600 opacity-30 blur-[130px] max-lg:top-0 lg:bottom-0 lg:translate-y-1/2"></div>
         <div class="absolute right-0 aspect-video w-[670px] translate-x-1/2 translate-y-1/2 rounded-full bg-secondary-100 opacity-30 blur-[130px] max-lg:bottom-0 lg:top-0 lg:-translate-y-1/2"></div>
-        <div class="relative flex flex-wrap flex-col lg:flex-row gap-8 justify-center">
-            @foreach($cards as $card)
+        <div class="relative flex flex-col flex-wrap justify-center gap-8 lg:flex-row">
```
Comment (Roene-JustBetter) [inline resources/views/page-builder/featured_cards.blade.php:15]: `max-lg:flex-col`

**Review by Roene-JustBetter — CHANGES_REQUESTED** [review]

**Review by Roene-JustBetter — DISMISSED** [review]

File: resources/views/page-builder/blog_overview.blade.php [hunk 5]
```diff
@@ -1,5 +1,5 @@
 @php($blogs = \Statamic\Statamic::tag('collection:blog')->fetch())
-<div class="relative py-16">
+<div class="relative py-16 component blog-overview">
```
Comment (jordythevulder) [inline resources/views/page-builder/blog_overview.blade.php:2]: should the component class be positioned first, since everything else is tailwind classes?

**Review by Roene-JustBetter — DISMISSED** [review]

**Comment by royduin** [comment]
So @Roene-JustBetter and @jordythevulder, frontend is the greatest we've ever built? And maybe from the Statamic end, any tips @BobWez98 or @kevinmeijer97? Everything is looking good?

File: resources/fieldsets/call_to_action.yaml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      display: 'Title - Heading'
+      icon: text
+      listable: hidden
+      instructions_position: above
+      visibility: visible
+      replicator_preview: true
+      hide_display: false
+      width: 50
+  -
+    handle: title_subheading
```
Comment (BobWez98) [inline resources/fieldsets/call_to_action.yaml:20]: I don't see the relevance of the `title_` prefix on heading and subheading but that's just an opinion

File: resources/fieldsets/faq.yaml [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+      display: 'Title - Heading'
+      icon: text
+      listable: hidden
+      instructions_position: above
+      visibility: visible
+      replicator_preview: true
+      hide_display: false
+      width: 50
+  -
+    handle: title_subheading
```
Comment (BobWez98) [inline resources/fieldsets/faq.yaml:20]: I see some duplication in this and `call_to_action.yaml` maybe make a fieldset?

File: resources/fieldsets/featured_cards.yaml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      type: text
+      display: 'Title - Heading'
+      icon: text
+      listable: hidden
+      instructions_position: above
+      visibility: visible
+      replicator_preview: true
+      hide_display: false
+  -
+    handle: title_subheading
```
Comment (BobWez98) [inline resources/fieldsets/featured_cards.yaml:17]: More duplication

File: resources/fieldsets/hero_banner.yaml [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+      display: Heading
+      icon: text
+      listable: hidden
+      instructions_position: above
+      visibility: visible
+      replicator_preview: true
+      hide_display: false
+      width: 50
+  -
+    handle: subheading
```
Comment (BobWez98) [inline resources/fieldsets/hero_banner.yaml:42]: I see a duplication here with `call_to_action.yaml` and `faq.yaml`

File: resources/fieldsets/partners.yaml [hunk 10]
```diff
@@ -0,0 +1,216 @@
+title: Partners
+fields:
+  -
+    import: component_settings
+  -
+    handle: title_heading
```
Comment (BobWez98) [inline resources/fieldsets/partners.yaml:6]: Duplication

File: resources/fieldsets/showcase_overview.yaml [hunk 11]
```diff
@@ -0,0 +1,28 @@
+title: 'Showcase overview'
+fields:
+  -
+    handle: title_heading
```
Comment (BobWez98) [inline resources/fieldsets/showcase_overview.yaml:4]: Duplication

File: resources/fieldsets/text_columns.yaml [hunk 12]
```diff
@@ -0,0 +1,120 @@
+title: 'Text columns'
+fields:
+  -
+    import: component_settings
```
Comment (BobWez98) [inline resources/fieldsets/text_columns.yaml:4]: Duplication

File: resources/fieldsets/text_columns.yaml [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+                    type: text
+                    display: Title
+                    icon: text
+                    listable: hidden
+                    instructions_position: above
+                    visibility: visible
+                    replicator_preview: true
+                    hide_display: false
+                -
+                  handle: text
```
Comment (BobWez98) [inline resources/fieldsets/text_columns.yaml:64]: I see this field a lot as well, if we want the same configuration like buttons and all the other stuff, why no fieldset if we ever want to change it?

File: resources/views/page-builder/faq.blade.php [hunk 14]
```diff
@@ -1,20 +1,20 @@
-<div class="relative z-10 bg-gray-100 pt-16 sm:pt-32"{{ ($component_id ?? false) ? " id=" . $component_id : "" }}>
-    <div class="max-w-7xl mx-auto px-6 sm:px-6 w-full z-10 relative">
-        @if(($title_heading ?? false) || ($title_subheading ?? false))
-            <h2 class="text-3xl font-extrabold tracking-tight sm:text-4xl text-center">
-                @if($title_heading ?? false)
-                    <span class="bg-clip-text text-transparent bg-gradient-to-b from-primary-100 to-primary-200 pb-2">{{ $title_heading }}</span>
+<div class="faq relative z-10 bg-gray-100 component pt-16 sm:pt-32"{{ $component_id ?? false ? ' id=' . $component_id : '' }}>
```
Comment (BobWez98) [inline resources/views/page-builder/faq.blade.php:1]: I see the custom classes scattered in the `class` attribute in some components, don't we want that grouped in either the front or end of the attribute? Happens in a few components

File: resources/views/page-builder/image_text.blade.php [hunk 15]
```diff
@@ -1,4 +1,4 @@
-<div class="mx-auto mb-12 flex max-w-7xl flex-col gap-20 px-6 md:mb-32 md:gap-28"{{ ($component_id ?? false) ? " id=" . $component_id : "" }}>
+<div class="mx-auto flex max-w-7xl flex-col gap-20 px-6 component md:gap-28"{{ ($component_id ?? false) ? " id=" . $component_id : "" }}>
```
Comment (BobWez98) [inline resources/views/page-builder/image_text.blade.php:1]: Maybe we can just do `id="$component_id ?? uniqueid()"` There might be a random `0` in html attributes now.

**Review by BobWez98 — CHANGES_REQUESTED** [review]

File: resources/fieldsets/call_to_action.yaml (hunk 6 above)
Comment (kevinmeijer97) [reply resources/fieldsets/call_to_action.yaml:20]: On the other hand, i'd rather see title & subtitle. The _heading is usually used for choosing actual heading tags on titles etc.

File: resources/fieldsets/call_to_action.yaml [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+        extra_button:
+          display: 'Extra button'
+          instructions: null
+          icon: null
+          sets:
+            button:
+              display: Button
+              instructions: null
+              icon: null
+              fields:
```
Comment (kevinmeijer97) [inline resources/fieldsets/call_to_action.yaml:68]: There's a couple of places where you've manually made the same button, maybe put it in a fieldset?

File: resources/fieldsets/text_columns.yaml (hunk 13 above)
Comment (kevinmeijer97) [reply resources/fieldsets/text_columns.yaml:64]: Agreed, i'd like to see a `text` fieldset apart from the `content` fieldset that already exists.

**Review by kevinmeijer97 — CHANGES_REQUESTED** [review]

File: resources/views/page-builder/faq.blade.php (hunk 14 above)
Comment (jordythevulder) [reply resources/views/page-builder/faq.blade.php:1]: Yeah i left the same comment, but it has been marked as resolved, but i don't see anything being resolved. [link](https://github.com/rapidez/website/pull/34#discussion_r1803099102)

File: resources/views/components/blog.blade.php [hunk 17]
```diff
@@ -1,13 +1,13 @@
...
         @responsive($image, ['class' => 'rounded-lg object-cover', 'loading' => 'lazy'])
     @endif
     <div class="flex flex-col self-center">
-        @if($category ?? false)
+        @if($category)
             <span class="text-sm text-secondary-100">{{ $category }}</span>
         @endif
-        @if($title ?? false)
+        @if($title)
                 <h3 class="mt-2 text-2xl font-semibold text-cyan-900">{{ $title }}</h3>
```
Comment (Roene-JustBetter) [inline resources/views/components/blog.blade.php:11]: Indenting

File: resources/views/components/card/project.blade.php [hunk 18]
```diff
@@ -1,10 +1,10 @@
 @props(['image', 'title'])
 <div class="w-full lg:w-1/3 flex flex-col justify-around rounded-2xl border border-gray-200 bg-white/70 px-6 py-8">
-    @if($image ?? false)
+    @if($image)
```
Comment (Roene-JustBetter) [inline resources/views/components/card/project.blade.php:3]: Here you don't use spaces between`@if()`

File: resources/views/page-builder/call_to_action.blade.php [hunk 19]
```diff
@@ -1,34 +1,35 @@
-<div class="mt-0 sm:-mt-16 relative z-20">
-    <div class="max-w-7xl mx-auto py-12 px-4 mb-10 text-center">
-        @if($logo ?? false)
-            <div class="flex items-center justify-center mb-10">
+<div class="component relative z-20">
+    <div class="mx-auto mb-10 max-w-7xl px-4 py-12 text-center">
+        @if ($logo)
```
Comment (Roene-JustBetter) [inline resources/views/page-builder/call_to_action.blade.php:3]: And here you use spaces between `@if ()`. So maybe use spaces everywhere.

File: resources/views/page-builder/hero-banner/default.blade.php [hunk 20]
```diff
@@ -1,54 +1,71 @@
...
-                        <a class="block" href="{{ $brand->social_twitter_url }}" target="_blank">
-                            <x-icon-twitter class="w-10 h-10"/>
+                <div class="mt-8 flex flex-wrap items-center gap-6 text-white">
+                    @if ($brand->social_twitter_url)
+                        <a
+                            class="block"
+                            href="{{ $brand->social_twitter_url }}"
+                            target="_blank"
+                        >
+                            <x-icon-twitter class="h-10 w-10" />
```
Comment (Roene-JustBetter) [inline resources/views/page-builder/hero-banner/default.blade.php:31]: Can you use `size-10` here?

File: resources/views/page-builder/hero-banner/default.blade.php [hunk 21]
```diff
@@ -1,54 +1,71 @@
...
-                    @if($brand->social_github_url ?? false)
-                        <a class="block" href="{{ $brand->social_github_url }}" target="_blank">
-                            <x-icon-github class="w-8 h-8"/>
+                    @if ($brand->social_github_url)
+                        <a
+                            class="block"
+                            href="{{ $brand->social_github_url }}"
+                            target="_blank"
+                        >
+                            <x-icon-github class="h-8 w-8" />
```
Comment (Roene-JustBetter) [inline resources/views/page-builder/hero-banner/default.blade.php:40]: Same here

File: resources/views/page-builder/partners.blade.php [hunk 22]
```diff
@@ -1,31 +1,24 @@
...
-    @if($columns ?? false)
+    @if ($columns)
         <div class="relative mt-12 flex w-full flex-wrap items-start justify-center overflow-hidden pb-5">
-            @foreach($columns as $column)
+            @foreach ($columns as $column)
                 <x-package.group :checked="$loop->first">
-                    @if($column->title ?? false)
-                        <x-slot name="label" class="{{ $loop->first ? 'rounded-l-lg' :  ($loop->last ? 'rounded-r-lg' : '')}}">{{ $column->title }}</x-slot>
+                    @if ($column->title)
+                        <x-slot name="label" class="{{ $loop->first ? 'rounded-l-lg' : ($loop->last ? 'rounded-r-lg' : '') }}">{{ $column->title }}</x-slot>
```
Comment (Roene-JustBetter) [inline resources/views/page-builder/partners.blade.php:17]: Do you need the `$loop->first` and `$loop->last`? You can't use `first: & last:`? https://tailwindcss.com/docs/hover-focus-and-other-states#first-last-odd-and-even

File: resources/views/partials/content/button.blade.php [hunk 23]
```diff
@@ -1,3 +1,7 @@
-@if(($button_link ?? false) && ($button_text ?? false))
-    <x-button href="{{ $button_link }}" target="{{ isset($open_in_new_tab) && $open_in_new_tab?->value() ? '_blank' : '_self' }}" class="mt-8">{{ $button_text }}</x-button>
+@if ($button_link && $button_text)
+    <x-button
+        href="{{ $button_link }}"
+        target="{{ isset($open_in_new_tab) && $open_in_new_tab?->value() ? '_blank' : '_self' }}"
+        class="mt-8"
+    >{{ $button_text }}</x-button>
```
Comment (Roene-JustBetter) [inline resources/views/partials/content/button.blade.php:6]: Indenting look a bit strange here.

File: resources/views/partials/nav.blade.php [hunk 24]
```diff
@@ -26,22 +30,26 @@ class="hidden items-center justify-center whitespace-nowrap rounded-full border
...
-                <button type="button" class="inline-flex items-center justify-center whitespace-nowrap rounded-md border border-transparent bg-white px-4 py-3 text-base text-primary-100 shadow-sm" aria-expanded="false">
+            <div class="ml-2 sm:ml-3 lg:hidden" @click="mobile_menu=!mobile_menu">
+                <button
+                    type="button"
+                    class="inline-flex items-center justify-center whitespace-nowrap rounded-md border border-transparent bg-white px-4 py-3 text-base text-primary-100 shadow-sm"
+                    aria-expanded="false"
+                >
                     <span class="sr-only">Open menu</span>
-                    <x-icon name="heroicon-o-menu" class="h-6 w-6"/>
+                    <x-icon name="heroicon-o-menu" class="h-6 w-6" />
```
Comment (Roene-JustBetter) [inline resources/views/partials/nav.blade.php:40]: `size-6`?

**Review by Roene-JustBetter — CHANGES_REQUESTED** [review]

**Review by BobWez98 — APPROVED** [review]

**Review by jordythevulder — DISMISSED** [review]

File: composer.lock
Comment (royduin) [inline composer.lock:1]: Why is this file removed?

**Review by royduin — CHANGES_REQUESTED** [review]
